Add C# interop_matrix v1.0.1 tests

jtattermusch opened the pull request:
I locally patched the v1.0.1 interop images and pushed to GCR. Now both csharp and csharpcoreclr build fine.
csharpcoreclr tests are passing, I'm still getting some errors for csharp, but as far as I can tell, they are also present when building C# v1.0.1 from scratch manually. (so they don't have to do with the interop_matrix framework).

```
Unhandled Exception:
System.ArgumentNullException: Value cannot be null.
Parameter name: source
  at System.Linq.Enumerable.Select[TSource,TResult] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`3[T1,T2,TResult] selector) [0x0000d] in <521f3fe476f34ea88748e97b3b984d05>:0 
  at CommandLine.Core.Tokenizer.ExplodeOptionList (RailwaySharp.ErrorHandling.Result`2[TSuccess,TMessage] tokenizerResult, System.Func`2[T,TResult] optionSequenceWithSeparatorLookup) [0x00014] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Core.Tokenizer+<>c__DisplayClass5_0.<ConfigureTokenizer>b__0 (System.Collections.Generic.IEnumerable`1[T] arguments, System.Collections.Generic.IEnumerable`1[T] optionSpecs) [0x00083] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Parser.Tokenize (System.Collections.Generic.IEnumerable`1[T] arguments, System.Collections.Generic.IEnumerable`1[T] optionSpecs, CommandLine.ParserSettings settings) [0x00017] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Parser.<ParseArguments>b__11_0[T] (System.Collections.Generic.IEnumerable`1[T] arguments, System.Collections.Generic.IEnumerable`1[T] optionSpecs) [0x00000] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Core.InstanceBuilder+<>c__DisplayClass0_0`1[T].<Build>b__7 () [0x00005] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Core.InstanceBuilder.Build[T] (CSharpx.Maybe`1[T] factory, System.Func`3[T1,T2,TResult] tokenizer, System.Collections.Generic.IEnumerable`1[T] arguments, System.StringComparer nameComparer, System.Boolean ignoreValueCase, System.Globalization.CultureInfo parsingCulture, System.Collections.Generic.IEnumerable`1[T] nonFatalErrors) [0x0015a] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Parser.ParseArguments[T] (System.Collections.Generic.IEnumerable`1[T] args) [0x00075] in <3a51da4303d24d9087548524f226a533>:0 
  at Grpc.IntegrationTesting.InteropClient.Run (System.String[] args) [0x00006] in <7769ed40adc44dc3a2895bd00dd0ea48>:0 
  at Grpc.IntegrationTesting.Client.Program.Main (System.String[] args) [0x00001] in <53e672eeb86441af8251ff4286f59d21>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentNullException: Value cannot be null.
Parameter name: source
  at System.Linq.Enumerable.Select[TSource,TResult] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`3[T1,T2,TResult] selector) [0x0000d] in <521f3fe476f34ea88748e97b3b984d05>:0 
  at CommandLine.Core.Tokenizer.ExplodeOptionList (RailwaySharp.ErrorHandling.Result`2[TSuccess,TMessage] tokenizerResult, System.Func`2[T,TResult] optionSequenceWithSeparatorLookup) [0x00014] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Core.Tokenizer+<>c__DisplayClass5_0.<ConfigureTokenizer>b__0 (System.Collections.Generic.IEnumerable`1[T] arguments, System.Collections.Generic.IEnumerable`1[T] optionSpecs) [0x00083] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Parser.Tokenize (System.Collections.Generic.IEnumerable`1[T] arguments, System.Collections.Generic.IEnumerable`1[T] optionSpecs, CommandLine.ParserSettings settings) [0x00017] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Parser.<ParseArguments>b__11_0[T] (System.Collections.Generic.IEnumerable`1[T] arguments, System.Collections.Generic.IEnumerable`1[T] optionSpecs) [0x00000] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Core.InstanceBuilder+<>c__DisplayClass0_0`1[T].<Build>b__7 () [0x00005] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Core.InstanceBuilder.Build[T] (CSharpx.Maybe`1[T] factory, System.Func`3[T1,T2,TResult] tokenizer, System.Collections.Generic.IEnumerable`1[T] arguments, System.StringComparer nameComparer, System.Boolean ignoreValueCase, System.Globalization.CultureInfo parsingCulture, System.Collections.Generic.IEnumerable`1[T] nonFatalErrors) [0x0015a] in <3a51da4303d24d9087548524f226a533>:0 
  at CommandLine.Parser.ParseArguments[T] (System.Collections.Generic.IEnumerable`1[T] args) [0x00075] in <3a51da4303d24d9087548524f226a533>:0 
  at Grpc.IntegrationTesting.InteropClient.Run (System.String[] args) [0x00006] in <7769ed40adc44dc3a2895bd00dd0ea48>:0 
  at Grpc.IntegrationTesting.Client.Program.Main (System.String[] args) [0x00001] in <53e672eeb86441af8251ff4286f59d21>:0 
FAILED: csharp__csharp_v1.0.1:csharp:grpc-test4:client_streaming [ret=1, pid=196278, time=1.0sec]
```

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

jtattermusch commented:
I think they were, but it's >1 years back so I'm not sure to be honest
(could have been some last minute breakage we didn't notice at that time).
Also it might have to do with mono runtime versions and/or nuget package
versions (both changed many times since then).

On Thu, Nov 16, 2017 at 7:03 PM, adelez <notifications@github.com> wrote:

> Thank you for creating the build but we cannot enable it if tests are
> failing. Do you know where this bug exist? In the client or in the
> integration with prod? I suppose they passed when we released 1.0.1?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc/pull/13416#issuecomment-345007208>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJeq5NBreFEqJGu7-sbnHCN-KCoOXmp-ks5s3HjugaJpZM4QgYlq>
> .
>



-- 

Jan

adelez commented:
Sorry for the long delay. I totally missed this PR.

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

jtattermusch commented:
C#  interop tests are now passing on v1.0.1, but they hang on the shutdown it seems (the csharpcoreclr tests are passing though).
Not sure if I can do anything about it.

```
running server_streaming
Passed!
TIMEOUT: csharp__csharp_v1.0.1:csharp:grpc-test4:server_streaming [pid=27905, time=60.0sec]

running cancel_after_begin
Passed!
TIMEOUT: csharp__csharp_v1.0.1:csharp:grpc-test4:cancel_after_begin [pid=27915, time=60.0sec]

running cancel_after_first_response
Passed!
TIMEOUT: csharp__csharp_v1.0.1:csharp:grpc-test4:cancel_after_first_response [pid=27928, time=60.0sec]

running timeout_on_sleeping_server
Passed!
TIMEOUT: csharp__csharp_v1.0.1:csharp:grpc-test4:timeout_on_sleeping_server [pid=27938, time=60.0sec]
FAILED: Some tests failed
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

jtattermusch commented:
@adelez   I rewrote the PR, PTAL

A few remarks:
- I've added a patch that enables building the C# v1.0.1 interop images.
- I realized that the `create_matrix_images.py` and  `tools/interop_matrix/run_interop_matrix_tests.py`  are not intuitive to use and also would really need some refactoring (the code is way to complex for what it's doing) - I made a few changes and also added comments to places in the code where refactoring would be useful.
- I've rebuilt and uploaded C# v1.0.1 images.

Currently, both csharp and csharpcoreclr v1.0.1 tests are passing reliably if I use `--server_host grpc-test.sandbox.googleapis.com`, but I'm getting some errors when running with `--server_host 74.125.206.210`
(which is the default setting) - can you please retest and advise what to do next? There's really nothing more I can do on the C# side to make these pass I believe.

this one passes reliably:
`tools/interop_matrix/run_interop_matrix_tests.py -l csharp --release v1.0.1 --server_host grpc-test.sandbox.googleapis.com`

this one shows some errors:
`tools/interop_matrix/run_interop_matrix_tests.py -l csharp --release v1.0.1`

grpc-testing commented:
```
[trickle] No significant performance differences
```

jtattermusch commented:
I tried running `tools/interop_matrix/run_interop_matrix_tests.py -l csharp --release v1.1.4` locally and I'm getting the same errors as with v1.0.1. Because v1.1.4 is generally green on the kokoro grpc_interop_matrix suite, I assume it's something about my testing environment  and C# v1.0.1 test should also pass on kokoro one this PR gets merged.

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

adelez commented:
LGTM
